include/gnc_attitude_estimation.hpp

```diff
@@ -0,0 +1,72 @@
+//
+// include/gnc_attitude_estimation.hpp
+// PSim
+//
+// Contributors:
+//   Kyle Krol  kpk63@cornell.edu
+//
+// Pathfinder for Autonomous Navigation
+// Space Systems Design Studio
+// Cornell Univeristy
+//
+
+#ifndef PAN_PSIM_INCLUDE_GNC_ATTITUDE_ESTIMATION_HPP_
+#define PAN_PSIM_INCLUDE_GNC_ATTITUDE_ESTIMATION_HPP_
+
+#include "gnc_containers.hpp"
+#include "lin.hpp"
+
+namespace gnc {
+
+/** @struct AttitudeEstimatorState
+ *  Contains internal state information needed by the estimate_attitude
+ *  function. */
+struct AttitudeEstimatorState {
+  double t;
+  lin::Vector4f q_body_eci;
+  CircularBuffer<lin::Vector3f, 5> w_buffer;
+  /** Defaults everything's value to NaN. */
+  AttitudeEstimatorState();
+};
+
+/** @struct AttitudeEstimatorData
+ *  Essentially serves as the inputs to the estimate_attitude function. See
+ *  member documentation for more details. */
+struct AttitudeEstimatorData {
+  /** Timestamp for the rest of the data in this struct and the time for which
+   *  an attitude estimate will be calculated. This should be in seconds since
+   *  the 'PAN epoch'. */
+  double t;
+  /** Position of the satellite in ECEF (units m). */
+  lin::Vector3d r_ecef;
+  /** Magnetic field vector in the body frame of the spacecraft (units T). */
+  lin::Vector3f b_body;
+  /** Sun vector in the body frame of the spacecraft (unit vector). */
+  lin::Vector3f s_body;
```

Comment:
Don't we also need sun sensor vector validity here?

Reply:
Sun vector validity (and validity for all measurements in fact) will just be conveyed by whether or not a field in `nan`. That's why all the constructors default all values to `nan` or in the case of the buffer empty.

Reply:
That's clever!